.github/workflows/smoke-test-release.yml

```diff
@@ -1,185 +1,690 @@
 name: Smoke test release
 on:
+    release:
+        types: [published]
     workflow_dispatch:
         inputs:
-            release_id:
-                description: 'WooCommerce Release Id'
+            tag:
+                description: 'WooCommerce Release Tag'
                 required: true
-
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
 permissions: {}
+env:
+    E2E_WP_LATEST_ARTIFACT: e2e-wp-latest--run-${{ github.run_number }}
+    E2E_UPDATE_WC_ARTIFACT: e2e-update-wc--run-${{ github.run_number }}
+    FORCE_COLOR: 1
 
 jobs:
-    login-run:
-        name: Daily smoke test on release.
+    get-tag:
+        name: Get WooCommerce release tag
+        permissions:
+            contents: read
         runs-on: ubuntu-20.04
+        outputs:
+            tag: ${{ steps.tag.outputs.result }}
+            created: ${{ steps.created-at.outputs.created }}
+        steps:
+            - name: Validate tag
+              if: ${{ github.event_name == 'workflow_dispatch' }}
+              env:
+                  GH_TOKEN: ${{ github.token }}
+              run: gh release view "${{ github.event.inputs.tag }}" --repo=woocommerce/woocommerce
+
+            - name: Get tag
+              uses: actions/github-script@v5
+              id: tag
+              with:
+                  result-encoding: string
+                  script: |
+                      console.log( "${{ github.event_name }}" );
+                      return "${{ github.event.release.tag_name }}" || "${{ github.event.inputs.tag }}"
+
+            - name: Verify woocommerce.zip asset
+              env:
+                  GITHUB_TOKEN: ${{ github.token }}
+              run: |
+                  gh release download ${{ steps.tag.outputs.result }} --repo woocommerce/woocommerce
+                  if [[ -f "woocommerce.zip" ]]
+                    then
+                      echo "${{ github.event.inputs.tag }} has a valid woocommerce.zip asset."
+                      exit 0
+                  fi
+
+                  echo "${{ github.event.inputs.tag }} does not have a valid woocommerce.zip asset."
+                  exit 1
+
+            - name: Get 'created-at' of WooCommerce zip
+              id: created-at
+              env:
+                  GH_TOKEN: ${{ github.token }}
+              run: echo "created=$(gh release view ${{ steps.tag.outputs.result }} --json assets --jq .assets[0].createdAt --repo woocommerce/woocommerce)" >> $GITHUB_OUTPUT
+
+    e2e-update-wc:
```

Comment:
After determining which release version to use, the very first test is to check whether our Bluehost release smoke test site will be able to successfully update to it by  running only the `update-woocommerce.spec.js` test.

Failure means the API and the rest of the E2E tests will not be executed anymore (the `api-wp-latest` and `e2e-wp-latest` jobs respectively), but the Allure report for the failing  `update-woocommerce.spec.js` test would still be published so we could still investigate why it failed.